rllib/models/modelv2.py

```diff
@@ -439,7 +439,16 @@ def _unpack_obs(obs: TensorType, space: Space, tensorlib: Any = tf) -> TensorStr
             )
         offset = 0
         if tensorlib == tf:
-            batch_dims = [v if isinstance(v, int) else v.value for v in obs.shape[:-1]]
+
+            def get_value(v):
+                if v is None:
+                    return -1
+                elif isinstance(v, int):
+                    return v
+                else:
+                    return v.value
+
+            batch_dims = [get_value(v) for v in obs.shape[:-1]]
             batch_dims = [-1 if v is None else v for v in batch_dims]
```

Comment:
then you don't need this line anymore. please delete?

Reply:
In fact, v.value can still be None:
```python

import tensorflow.compat.v1 as tf
tf.disable_v2_behavior()

obs = tf.placeholder(tf.float32, shape=[None, None, 128, 3], name="X")
print(obs.shape) 
>>> TensorShape([Dimension(None), Dimension(None), Dimension(128), Dimension(3)])

def get_value(v):
    if v is None:
        return -1
    elif isinstance(v, int):
        return v
    else:
        return v.value 

batch_dims = [get_value(v) for v in obs.shape[:-1]]
print(batch_dims)
>>> [None, None, 128]
batch_dims = [-1 if v is None else v for v in batch_dims]
print(batch_dims)
>>> [-1, -1, 128]
```
But still, this logic can be moved in get_value: bd0e75130a53e6024b10390a3d705432a21c135b

Reply:
Checking `v.value is None` before `isinstance(v, int)` will lead to something like "ValueError, int type does not have a .value method" if v is an integer. Can change:
```python
def get_value(v):
  if v is None:
      return -1
  elif isinstance(v, int):
      return v
  elif v.value is None:
      return -1
  else:
      return v.value
```

to:
```python
def get_value(v):
  if isinstance(v, int):
      return v
  elif  v is None or v.value is None:
      return -1
  else:
      return v.value

```

But in general, I prefere to avoid unnecessary calls to .isinstance because this function can be quite slow.